trinity/config.py

```diff
@@ -457,14 +500,13 @@ def node_class(self) -> Type['Node']:
         from trinity.nodes.full import FullNode
         from trinity.nodes.light import LightNode
 
-        if self.is_full_mode:
+        if self.base_config.is_full_mode:
```

Comment:
I moved the `node_class` into the `TrinityEth1SubConfig` because I personally think it is unclear at this point if the beacon node will actually end up with a concept of a `Node` class and even if we do it may be totally different from the existing one. More on that here: https://github.com/ethereum/py-evm/pull/1536#issuecomment-443317636

Maybe the same could be said about `get_chain_config` but I would defer that decision to a later point when we get our hands on that.